### PR TITLE
3rd-party BOM, no dep versions in project

### PR DESCRIPTION
**Pull request by akomakom**
* To get around BOM limitations, added `gmavenplus` code that dynamically determines versions that are used in plugin configs.
* How can I check whether all my changes to `<provided>` (moving them across levels since I'm getting rid of a lot of `<dependencyManagement>` content) did not screw anything up?

Azure builds will fail until dso PR is merged (and system-tests-parent is published)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -314,6 +169,51 @@
...
+                <scripts>
+                  <script><![CDATA[
+                    def find = {artifactId ->
+                      project.artifacts.find{it.artifactId == artifactId}?.version ?: ''
+                    }
+                    def props = [
+                            'slf4j.version'           : find('slf4j-api'),
+                            'hibernate-core.version'  : find('hibernate-core')
+                    ]
+                    println "Resolved depdendency versions: ${props}"
```
Comment (chrisdennis) [inline pom.xml]: `dependency`

**Review by chrisdennis — APPROVED** [review]
One minor typo, but I think the transformation as a whole makes sense. I wonder if it's worth in some way diff'ing the `dependency:tree` output to validate that the graph hasn't changed at all?

File: pom.xml (hunk 1 above)
Comment (akomakom) [reply pom.xml]: There are a few scope differences, will un-draft once fixed

**Comment by rjatkins** [comment]
When will a release with this change, and its associated security upgrades, be published to maven central? I can see that ehcache 2.10.10 was released on Oct 15 according to https://confluence.terracotta.org/display/release/Home but it's still not showing up in the terracotta maven repo, or in central.

**Comment by akomakom** [comment]
Thanks @rjatkins for bringing that up.  2.10.10 has now been released and should show up in maven central shortly.

**Comment by rjatkins** [comment]
Ah, it looks like what was tagged for 2.10.10 was done without any update to the version of management-core - it's still using 2.1.27 https://github.com/ehcache/ehcache2/blob/v2.10.10/pom.xml#L27 - and also without this PR included. I can see that both of these are fixed on the release/2.10.10 branch though https://github.com/ehcache/ehcache2/blob/release/2.10.10/pom.xml#L23,L26 What exact version are you going to release to central?

**Comment by rjatkins** [comment]
Wait, I see that mvnrepository.com has finally indexed ehcache 2.10.10.0.95 on the terracotta maven repo. And ah, that still doesn't have the jersey-server 2.35 upgrade in it. It seems I need a release of ehcache 2.10.10.x with thirdparty-bom-4.x 4.3.10.1.12 or newer https://github.com/Terracotta-OSS/terracotta-thirdparty-bom-4.x/blob/v4.3.10.1.12/pom.xml#L21 - like the recently tagged 2.10.10.1.13 https://github.com/ehcache/ehcache2/blob/v2.10.10.1.13/pom.xml#L18 Will that version make it to the terracotta maven repo or to central soon?

**Comment by akomakom** [comment]
The `2.10.10` OSS release that has yet to appear on central was done from the same commit as 2.10.10.0.95, essentially in lock-step with Terracotta Bigmemory lifecycle.  This is indeed from before the thirdparty-bom conversion.

We have yet to work out a strategy for OSS releases after the conversion, as they will require releases of management-common and thirdparty-bom-4.x.

**Pull request merged**